### PR TITLE
Thread safe add() for Cache Repository for the MemcachedStore

### PR DESCRIPTION
**Pull request by winglian**
The current Cache Repository implementation of add() uses if (!get then put) which is not really thread safe. It is probably safer to use Memcached's add method directly.

### CODE REVIEW

File: src/Illuminate/Contracts/Cache/ThreadSafeStore.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	 * Store an item in the cache if the key does not exist for a given number of minutes.
+	 *
+	 * @param  string  $key
+	 * @param  mixed   $value
+	 * @param  int     $minutes
+	 * @return bool
+	 */
+	public function add($key, $value, $minutes);
+
+}
```
Comment (GrahamCampbell) [inline src/Illuminate/Contracts/Cache/ThreadSafeStore.php]: Missing new line.

**Comment by MaartenStaa** [comment]
This is much better than what Laravel does by default :thumbsup:

**Comment by GrahamCampbell** [comment]
I wonder if this would be ok to go into 5.0. Don't retarget this, but just wait for Taylor.

**Comment by taylorotwell** [comment]
Did a little differently on 5.0 branch.

**Pull request closed without merging**